tests/Properties.hs

```diff
@@ -51,6 +54,10 @@ prop_fromSingletonIndex_singletonIndex :: Int64 -> Bool
 prop_fromSingletonIndex_singletonIndex i =
     fromSingletonIndex (singletonIndex i) == Just i
 
+prop_foldable :: [(Int64, Int)] -> Bool
+prop_foldable xs = F.foldMap snd xs' == F.foldMap id (Tree.fromList xs')
+  where xs' = nubBy (\x y -> fst x == fst y) . map (\x -> (fst x, Sum $ snd x)) $ xs
+
```

Comment:
can you add this test case:

prop_toList_fromList :: [(Int64, Int)] -> Bool
prop_toList_fromList xs =
    F.toList (Tree.fromList xs) ==
    F.toList (M.fromList xs)

and if this fails, change the order in Tree.fromList